pkg/util/trivy/scanner.go

```diff
@@ -0,0 +1,213 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build trivy
+// +build trivy
+
+package trivy
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/sbom"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/workloadmeta"
+
+	cutil "github.com/DataDog/datadog-agent/pkg/util/containerd"
+	"github.com/containerd/containerd"
+)
+
+const (
+	imagesToScanBufferSize = 50
+	fsToScanBufferSize     = 1
+)
+
+var _service *Scanner
+
+type scanRequest struct {
+	scanOptions sbom.ScanOptions
+	onSuccess   sbom.ScanSuccessCallback
```

Comment:
The issue with callbacks is that you remove control flow execution from the caller, it enforces the callbacks to be called in separate goroutine (not necessary) or in Scanner goroutine (may block the Scanner forever).
In this scenario, it may more suited to ask for a reply channel with a `scanResult` object, it also makes it easier to wait for scan result.
WDYT?